01-data-structures/02-stacks-and-queues/myqueue/myqueue.rb

```diff
@@ -5,14 +5,25 @@ class MyQueue
   def initialize
     @queue = Array.new
     @head = @queue[0]
+		@tail = @queue[-1] # added by Tom
   end
 
   def enqueue(element)
+		@queue.insert(0, element)
+		@head = @queue[0]
+		@tail = @queue[-1]
   end
 
   def dequeue
+		@queue.delete(@queue[-1])
+		@tail = @queue[-1]
   end
 
   def empty?
+		if @queue.length == 0
+			return true
+		else
+			return false
+		end
```

Comment:
Since this returns a boolean, it can be condensed to:

```ruby
def empty?
  @queue.length == 0
end
```

If the length is indeed 0, that comparison will result in `true`, thus returning `true` from the method. Likewise with `false`.